homeassistant/components/systemair/sensor.py

```diff
@@ -0,0 +1,505 @@
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_REHEATER_TEMP_SENSOR
+        ],
+        ATTR_ALARM_SUPPLY_TEMP_SENSOR: [
+            'Alarm Supply Temperature Sensor',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_SUPPLY_TEMP_SENSOR
+        ],
+        ATTR_ALARM_INDOOR_TEMP_SENSOR: [
+            'Alarm Indoor Temperature Sensor',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_INDOOR_TEMP_SENSOR
+        ],
+        ATTR_ALARM_EXTRACT_TEMP_SENSOR: [
+            'Alarm Extract Temperature Sensor',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_EXTRACT_TEMP_SENSOR
+        ],
+        ATTR_ALARM_PREHEATER_TEMP_SENSOR: [
+            'Alarm Preheater Temperature Sensor',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_PREHEATER_TEMP_SENSOR
+        ],
+        ATTR_ALARM_EFFICIENCY_TEMP_SENSOR: [
+            'Alarm Efficiency Temperature Sensor',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_EFFICIENCY_TEMP_SENSOR
+        ],
+        ATTR_ALARM_PDM_RH: [
+            'Alarm PDM Relative Humidity',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_PDM_RH
+        ],
+        ATTR_ALARM_PDM_TEMP: [
+            'Alarm PDM Temperature Sensor',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_PDM_TEMP
+        ],
+        ATTR_ALARM_CHANGE_FILTER: [
+            'Alarm Filter Replacement',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_CHANGE_FILTER
+        ],
+        ATTR_ALARM_EXTRA_CONTROLLER: [
+            'Alarm Extra Controller Malfunction',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_EXTRA_CONTROLLER
+        ],
+        ATTR_ALARM_EXTERNAL_STOP: [
+            'Alarm External Stop',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_EXTERNAL_STOP
+        ],
+
+        ATTR_ALARM_MANUAL_STOP: [
+            'Alarm Stopped Manually',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_MANUAL_STOP
+        ],
+
+        ATTR_ALARM_REHEATER_OVERHEAT: [
+            'Alarm Overheated Reheater',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_REHEATER_OVERHEAT
+        ],
+
+        ATTR_ALARM_CO2: [
+            'Alarm Co2 Quality',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_CO2
+        ],
+
+        ATTR_ALARM_RH: [
+            'Alarm Humidity Quality',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_RH
+        ],
+
+        ATTR_ALARM_INCORRECT_MANUAL_MODE: [
+            'Alarm Incorrect Manual Mode',
+            '',
+            'mdi:alert',
+            SENSOR_ALARM_INCORRECT_MANUAL_MODE
+        ],
+
+        ATTR_TARGET_TEMPERATURE: [
+            'Target Temperature',
+            TEMP_CELSIUS,
+            'mdi:alert',
+            SENSOR_TARGET_TEMPERATURE
+        ]
+
+    }
```

Comment:
Like before, no globals and constrain this out of setup function. Also some of this value looks more as a binary_sensor and not as sensor.

Reply:
To accommodate the standards of HA I adopted the implementation fromhttps://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/comfoconnect/sensor.py. it's quite hard to know how this should be implemented if there is other components that does not comply. Could you refer me to an implementation that comply with HA standards?